projects/ad4696_fmc/common/ad4696_bd.tcl

```diff
@@ -0,0 +1,123 @@
+
+create_bd_intf_port -mode Master -vlnv analog.com:interface:spi_master_rtl:1.0 ad4696_spi
+
+## To support the 1MSPS (SCLK == 80 MHz), set the spi clock to 160 MHz
+
+ad_ip_instance axi_clkgen spi_clkgen
+ad_ip_parameter spi_clkgen CONFIG.CLK0_DIV 5
+ad_ip_parameter spi_clkgen CONFIG.VCO_DIV 1
+ad_ip_parameter spi_clkgen CONFIG.VCO_MUL 8
+ad_connect spi_clkgen/clk sys_ps7/FCLK_CLK0
+ad_connect spi_clk spi_clkgen/clk_0
+  
+# create a SPI Engine architecture
+
+create_bd_cell -type hier spi_ad4696
+current_bd_instance /spi_ad4696
+
+  create_bd_pin -dir I -type clk clk
+  create_bd_pin -dir I -type rst resetn
+  create_bd_pin -dir I -type clk spi_clk
+  create_bd_pin -dir O irq
+  create_bd_intf_pin -mode Master -vlnv analog.com:interface:spi_master_rtl:1.0 m_spi
+  create_bd_intf_pin -mode Master -vlnv xilinx.com:interface:axis_rtl:1.0 M_AXIS_SAMPLE
+
+  if {$adc_resolution <= 32} {
+    set data_width 32
+  } else {
+    set data_width 64
+  };
+
+  ad_ip_instance spi_engine_execution execution
+  ad_ip_parameter execution CONFIG.DATA_WIDTH $data_width
+  ad_ip_parameter execution CONFIG.NUM_OF_CS 1
+  ad_ip_parameter execution CONFIG.NUM_OF_SDI 1
+  ad_ip_parameter execution CONFIG.SDO_DEFAULT 1
+  ad_ip_parameter execution CONFIG.SDI_DELAY 2
+
+  ad_ip_instance axi_spi_engine axi_regmap
+  ad_ip_parameter axi_regmap CONFIG.DATA_WIDTH $data_width
+  ad_ip_parameter axi_regmap CONFIG.NUM_OFFLOAD 1
+  ad_ip_parameter axi_regmap CONFIG.ASYNC_SPI_CLK 1
+
+  ad_ip_instance spi_engine_offload offload
+  ad_ip_parameter offload CONFIG.DATA_WIDTH $data_width
+  ad_ip_parameter offload CONFIG.ASYNC_SPI_CLK 1
+
+  ad_ip_instance spi_engine_interconnect interconnect
+  ad_ip_parameter interconnect CONFIG.DATA_WIDTH $data_width
+
+  ad_ip_instance util_pulse_gen trigger_gen
```

Comment:
I think we should use the axi_pulse_gen to set up the sampling rate. That way the software will be able to change it.